onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc

```diff
@@ -1141,15 +1142,16 @@ void PoolOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const Nod
   input_indices.push_back(model_builder.AddOperandFromScalar(kernel_shape[0]));
   input_indices.push_back(model_builder.AddOperandFromScalar(fuse_code));
 
-  // TODO support API 28
-  input_indices.push_back(model_builder.AddOperandFromScalar(use_nchw));
+  if (model_builder.GetAndroidSdkVer() > 28) {  // nchw only supported on api 29+
+    input_indices.push_back(model_builder.AddOperandFromScalar(use_nchw));
+  }
 
```

Comment:
If SDK is < 29 do we need to do transforms somewhere for it to work?

Reply:
GetMinSupportedSdkVer at line 1002 will specifically disable this operator if use_nchw is true and it's API 28

---
In reply to: [459170891](https://github.com/microsoft/onnxruntime/pull/4582#discussion_r459170891) [](ancestors = 459170891)